pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/CloseResourceRule.java

```diff
@@ -678,6 +679,11 @@ private void addCloseResourceViolation(ASTVariableId id, TypeNode type, Object d
     }
 
     private String getResourceTypeName(ASTVariableId varId, TypeNode type) {
+        if (type == null) {
```

Comment:
I wonder, whether this should be fixed somewhere else... type is null here, because we couldn't determine the type earlier in method "getTypeOfVariable" (I think, I'm just reading the code...).

Would we have a similar NPE with the following code snippet?

```java
package mypackage;

import java.io.Closeable;
import java.util.List;

public class CloseResourceCrasher {
    Closeable myCloseable;

    @Override
    public void close() {
        var c = myClosable;
        try {
            c.close();
        } catch (Exception e) { }
    }
}
```

hm... ok, here we would have a initializer expression for the var, which should take the type of the field. So, maybe this works fine.

Reply:
Alright. I was also not able to find a more elegant way. The rule is already quite difficult in the implementation - and the type at this location is only retrieved for the sake of reporting... (the runtime type might be different than the reference type of the variable).
Let's keep it as you solved it. One NPE less.